base/strings/util.jl

```diff
@@ -484,13 +486,15 @@ function rpad(
     s::Union{AbstractChar,AbstractString},
     n::Integer,
     p::Union{AbstractChar,AbstractString}=' ',
-) :: String
+) :: if any(isa.((s, p), Union{StyledString, StyledChar, SubString{<:StyledString}})) StyledString{String} else String end
```

Comment:
This is too complicated of a type assert.

Reply:
Huh, I just tested type inference and it seems like it's completely resolved (to a `String`/`StyledString` as appropriate), so I suppose no type assert is needed at all?